sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/StringExpressionsSuite.scala

```diff
@@ -464,6 +464,41 @@ class StringExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper {
     checkEvaluation(StringSpace(s1), null, row2)
   }
 
+  test("RegexReplace") {
+    val row1 = create_row("100-200", "(\\d+)", "num")
```

Comment:
Maybe we could add a new test suite to test expressions with mutable states(like this one and the Rand), focus on testing the behavior of the expression across multiply evaluations, and using literal here to test the functionality with one time evaluation.